server/src/main/java/org/eclipse/openvsx/entities/ExtensionVersion.java

```diff
@@ -137,24 +134,27 @@ public ExtensionJson toExtensionJson() {
             json.publishedBy = this.getPublishedWith().getUser().toUserJson();
         }
         if (this.getDependencies() != null) {
-            json.dependencies = CollectionUtil.map(this.getDependencies(), depExtension -> {
-                var ref = new ExtensionReferenceJson();
-                ref.namespace = depExtension.getNamespace().getName();
-                ref.extension = depExtension.getName();
-                return ref;
-            });
+            json.dependencies = toExtensionReferenceJson(this.getDependencies());
         }
         if (this.getBundledExtensions() != null) {
-            json.bundledExtensions = CollectionUtil.map(this.getBundledExtensions(), bndExtension -> {
-                var ref = new ExtensionReferenceJson();
-                ref.namespace = bndExtension.getNamespace().getName();
-                ref.extension = bndExtension.getName();
-                return ref;
-            });
+            json.bundledExtensions = toExtensionReferenceJson(this.getBundledExtensions());
         }
         return json;
     }
 
+    private List<ExtensionReferenceJson> toExtensionReferenceJson(List<String> extensionReferences) {
+        return extensionReferences.stream().map(fqn -> {
+            var split = fqn.split("\\.");
+            if (split.length != 2 || split[0].isEmpty() || split[1].isEmpty()) {
+                return null;
```

Comment:
Malformed extension references can only occur, if someone edited the DB after the extension was published. We perform sanity checks at the time of publishing.

Reply:
You could optimize it further and use `indexOf('.')` instead of `split` to avoid the effort of regular expression matching. Note that this method is invoked very often.

Reply:
done